submissions/eoneoff/data-structures-http/data_structures/linked_list.py

```diff
@@ -0,0 +1,44 @@
+from .node import Node
+
+class LinkedList:
+    def __init__(self):
+        self._root = None
+        self._length = 0
+
+    def __iter__(self):
+        current = self._root
+        while current:
+            yield current
+            current = current.next
+
+    def insert(self, value, successor = None):
+        if(successor and self._root.value != successor):
+            for node in self:

```

Comment:
Refer to `Common Data Structure Operations` table here: https://www.bigocheatsheet.com/
It is common that `insert` operation works for constant time. Your solution is O(n).
Please, reconsider implementation and improve working time of `insert` method.

Reply:
Sorry, but no. Insertion in single linked list is O(1) if we insert into the end of the list. But here the condition was to implement insertion before the successor, so we actually have two operations: searching for successor (which is O(n)) and then inserting (which is — after we have found a successor — is actually O(1)).

Reply:
You are right, but
1. insert and access are two different operations, and you can split your code into two operations
2. having successor of type Node, you already have all the data to perform insert operation, no need to perform a search first